records/2022-03-08.priority-lanes.md

```diff
@@ -0,0 +1,32 @@
+# Enforcing priority lanes
+
+Date: 2022-03-08
+
+## Status
+
+**DRAFT.**
+
+## Context
+
+Listing contextual elements to consider prior to offer suggestions and deepen our understanding of current limitations.
+
+### The database-tasks SQS queue
+
+Database operations are all sent through the database-tasks queue/worker. This has a benefit of lining up and tracking a batch of operations to reduce the load. A major downside is that this is a bottleneck for high priority database operations such as saving a notification that needs to be sent in the next seconds. There are use cases when GCNotify is constantly hit and the database-tasks queue grows bigger. There are no FIFO strategy set on this queue as we are targeting high volume processing. This means though that no priority distinction is possible for operations going through it.
+
+## Options
+
+Listed in this section are different options available to us to improve our priority lanes.
+
+### Improving database-tasks SQS queue
+
+* Making it faster
+* Getting high priority database operations out of it; but these need to be low volume.
```

Comment:
This could be an interesting challenge if people just switch everything over to high priority.  We'll need a way to validate that a service's high priority notifications are a valid use case.